### ISSUE TITLE
$PRELOAD$ string in preload list.

### ISSUE THREAD

**Issue by steevanb** [DarkGhostHunter/Preloader#32]
Hi,

I have this script to generate preload file:
```php
<?php

require('/var/www/phpbenchmarks/project/generator/var/generator/5e80978c2d086/generated/public/index.php');

$vendorDir = '/var/www/phpbenchmarks/project/generator/var/cache/dev/../../generator/5e80978c2d086/generated/vendor';
DarkGhostHunter\Preloader\Preloader::make()
    ->selfExclude()
    ->writeTo('/var/www/phpbenchmarks/project/generator/var/cache/dev/../../generator/5e80978c2d086/generated/.phpbenchmarks/php/7.4/preload.php');
```

Result contains `$PRELOAD$` in files list, i'm not sure it's needed ? ;)

```php
$files = [
    '/var/www/vendor/symfony/serializer/SerializerInterface.php',
    '$PRELOAD$',
    '/var/www/.phpbenchmarks/php/7.4/preload.php'
];
```

Another little bug: template have 2 empty lines at the end, so generated preload file contains this 2 lines. 1 should be enough ;)

**Comment by DarkGhostHunter** [comment]
Does it happen when you do not use self exclude?

**Comment by steevanb** [comment]
It always happen: with `selfExclude()`, `exclude()` or nothing else than `make()->writeTo()`.

**Comment by DarkGhostHunter** [comment]
Can you dump a list of your opcache cached list in the app? Seems weird that the `$PRELOAD$` appears in the list, since it's basically what Opcache is returning.

If you dump the list and you find `$PRELOADER$`, then it may be Opcache fault's.

Another thing crossing my mind is that Opcache may be including the preloader script location, which is later referenced internally as `$PRELOADER$` when opcache asks for the list.

**Comment by steevanb** [comment]
You're right, when i call `opcache_get_status()` i have this file in `scripts` section:

```
    ["$PRELOAD$"]=>
    array(6) {
      ["full_path"]=>
      string(9) "$PRELOAD$"
      ["hits"]=>
      int(0)
      ["memory_consumption"]=>
      int(560)
      ["last_used"]=>
      string(24) "Thu Jan  1 01:00:00 1970"
      ["last_used_timestamp"]=>
      int(0)
      ["timestamp"]=>
      int(0)
    }
```

**Comment by DarkGhostHunter** [comment]
I don't have it, so it may be something in your end, or as I suspect. Did some kind of preloading occur when you used `opcache_get_status()`? or just vanilla (no-preload) PHP?

**Comment by na3r** [comment]
Hi there, 
I have the same issue, I've used laraload to preload Laravel framework and in preload.php in $files array there is a $PRELOAD$ string that causes to throw an exception in preload file and php crashes on startup.  A quick fix can be to ignore invalid files in preload.php.stub on line 44

**Comment by DarkGhostHunter** [comment]
I think a better fix would be to filter any record that is equal to `$PRELOAD$`.

**Comment by na3r** [comment]
> I think a better fix would be to filter any record that is equal to `$PRELOAD$`.

but maybe there will be other strings and you should check all of them. But ignoring invalid files can prevent future problems i guess

**Comment by DarkGhostHunter** [comment]
> > I think a better fix would be to filter any record that is equal to `$PRELOAD$`.
> 
> but maybe there will be other strings and you should check all of them. But ignoring invalid files can prevent future problems i guess

What other strings? Preload is a weird beast, and I think it's better to strictly tell the developer he's preloading something that doesn't exists.

Until there is more information available (I haven't found anything in the PHP manual), I will manually exclude this string exclusively in the next patch version, coming hot today.

**Comment by DarkGhostHunter** [comment]
[Fixed in v2.0.5](https://github.com/DarkGhostHunter/Preloader/releases/tag/v2.0.5).
